lib/streamlit/elements/json.py

```diff
@@ -13,52 +13,28 @@
 # limitations under the License.
 
 import json
-import copy
 from typing import (
     Any,
-    Hashable,
     List,
-    MutableMapping,
-    Tuple,
     Union,
     cast,
     TYPE_CHECKING,
 )
 
-from typing_extensions import TypeAlias
-
 from streamlit.proto.Json_pb2 import Json as JsonProto
 from streamlit.runtime.state import SessionStateProxy
-from streamlit.type_util import is_iterable
 from streamlit.user_info import UserInfoProxy
 
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
 
-Body: TypeAlias = Union[List[Any], Tuple[Any, ...], MutableMapping[Hashable, Any]]
-
 
-def _convert_sets_to_lists(body: Body) -> Body:
-    if isinstance(body, (list, tuple)):
-        # We could technically iterate through the elements of a list/tuple and convert
-        # any sets that we find to lists like we do below, but lists/tuples of sets
-        # seem like a strange enough use-case that it's probably not worth the
-        # additional complexity.
-        return body
-
-    # Convert sets into lists, which render more nicely on the frontend
-    set_found = False
-    for key in body:
-        if isinstance(body[key], set):
-            if not set_found:
-                # When set is found to prevent mutation of input body, we need
-                # to shallow copy it once. To avoid copying it multiple times
-                # we use set_found flag.
-                body = copy.copy(body)
-                set_found = True
-            body[key] = list(body[key])
-    return body
+def _convert_sets_to_lists_repr(o: object) -> Union[str, List[Any]]:
```

Comment:
nit: since it is not only about converting sets to lists (and might extended to more structures in the future), maybe a name like `_make_serializable` (or `_ensure_serialization`) would be a slightly better fit.

Reply:
Sure 👍 Both suggestions have been added by me just now, and I'd like to thank you for your review 🥇